src/Microsoft.Azure.SignalR.AspNet/ClientConnections/ClientConnectionManager.cs

```diff
@@ -60,6 +62,8 @@ public ClientConnectionManager(HubConfiguration configuration, ILoggerFactory lo
 
         public bool TryAdd(string connectionId, IServiceConnection serviceConnection)
         {
+            TaskCompletionSource<bool> tcs = new TaskCompletionSource<bool>(TaskContinuationOptions.RunContinuationsAsynchronously);
```

Comment:
Should be `TaskCreationOptions`

Reply:
Why not add the tcs as a property of the Client Connection?

Reply:
Because we don't need to, and the param here is an interface so I don't want to make things more complex...

Reply:
Actually `TaskContinuationOptions.RunContinuationsAsynchronously` and `TaskCreationOptions.RunContinuationsAsynchronously` share the same value.

But I have fixed it.

Reply:
> Actually `TaskContinuationOptions.RunContinuationsAsynchronously` and `TaskCreationOptions.RunContinuationsAsynchronously` share the same value.


Just for future knowledge, just because they share the same value does not mean they are treated that same when creating a `TaskCompletionSource`. `TaskCompletionSource` has a constructor that takes an `object` and if you pass `TaskCompletionSource` in then it will use the `object` overload and end up being synchronous instead of the expected asynchronous behavior.